core/modules/l4_checksum.cc

```diff
@@ -49,28 +51,42 @@ void L4Checksum::ProcessBatch(Context *ctx, bess::PacketBatch *batch) {
     Ethernet *eth = batch->pkts()[i]->head_data<Ethernet *>();
 
     // Calculate checksum only for IPv4 packets
-    if (eth->ether_type != be16_t(Ethernet::Type::kIpv4))
+    if (eth->ether_type != be16_t(Ethernet::Type::kIpv4)) {
+      EmitPacket(ctx, batch->pkts()[i], DEFAULT_GATE);
       continue;
+    }
 
     Ipv4 *ip = reinterpret_cast<Ipv4 *>(eth + 1);
 
     if (ip->protocol == Ipv4::Proto::kUdp) {
       size_t ip_bytes = (ip->header_length) << 2;
       Udp *udp =
           reinterpret_cast<Udp *>(reinterpret_cast<uint8_t *>(ip) + ip_bytes);
-      udp->checksum = CalculateIpv4UdpChecksum(*ip, *udp);
+      if (verify)
+	 EmitPacket(ctx, batch->pkts()[i],
+		    (VerifyIpv4UdpChecksum(*ip, *udp)) ? DEFAULT_GATE : _DROP_GATE);
+      else {
+	 udp->checksum = CalculateIpv4UdpChecksum(*ip, *udp);
+	 EmitPacket(ctx, batch->pkts()[i], DEFAULT_GATE);
+      }
     } else if (ip->protocol == Ipv4::Proto::kTcp) {
       size_t ip_bytes = (ip->header_length) << 2;
       Tcp *tcp =
           reinterpret_cast<Tcp *>(reinterpret_cast<uint8_t *>(ip) + ip_bytes);
-      tcp->checksum = CalculateIpv4TcpChecksum(*ip, *tcp);
+      if (verify)
+	 EmitPacket(ctx, batch->pkts()[i],
+		    (VerifyIpv4TcpChecksum(*ip, *tcp)) ? DEFAULT_GATE : _DROP_GATE);
+      else
+	 tcp->checksum = CalculateIpv4TcpChecksum(*ip, *tcp);
     }
-
-    continue;
   }
-
-  RunNextModule(ctx, batch);
 }
-
+/*----------------------------------------------------------------------------------*/
+CommandResponse L4Checksum::Init(const bess::pb::L4ChecksumArg &arg) {
+  if (arg.verify())
+    verify = arg.verify();
```

Comment:
Assignment is fine here. Protobuf defaults are always the "zero" value when using these getters
```suggestion
verify = arg.verify();
```